src/main/java/org/datadog/jenkins/plugins/datadog/DatadogGlobalConfiguration.java

```diff
@@ -309,21 +343,105 @@ private FormValidation checkAgentConnectivity(final String host, final String po
      * @param targetApiKey - A String containing the apiKey submitted from the form on the
      *                   configuration screen, which will be used to authenticate a request to the
      *                   Datadog API.
+     * @param targetCredentialsApiKey - A String containing the API key as a credential, if it is not specified,
+                         try the connection with the targetApiKey
      * @return a FormValidation object used to display a message to the user on the configuration
      * screen.
      * @throws IOException      if there is an input/output exception.
      * @throws ServletException if there is a servlet exception.
      */
     @RequirePOST
-    public FormValidation doTestConnection(@QueryParameter("targetApiKey") final String targetApiKey, @QueryParameter("targetApiURL") final String targetApiURL)
+    public FormValidation doTestConnection(
+            @QueryParameter("targetApiKey") final String targetApiKey,
+            @QueryParameter("targetCredentialsApiKey") final String targetCredentialsApiKey, 
+            @QueryParameter("targetApiURL") final String targetApiURL)
             throws IOException, ServletException {
-        if (DatadogHttpClient.validateDefaultIntakeConnection(targetApiURL, Secret.fromString(targetApiKey))) {
+
+        Boolean validApiConnection = false;
+        StringCredentials credentialsApiKey = this.getCredentialFromId(targetCredentialsApiKey);
+        if (credentialsApiKey != null && !credentialsApiKey.getSecret().getPlainText().isEmpty()) {
+            validApiConnection = DatadogHttpClient.validateDefaultIntakeConnection(targetApiURL, credentialsApiKey.getSecret());
+        } else {
+            validApiConnection = DatadogHttpClient.validateDefaultIntakeConnection(targetApiURL, Secret.fromString(targetApiKey));
+        }
+        if (validApiConnection) {
```

Comment:
Have you considered to isolate the search of the `Secret` in a single method. You needn't to modify the rest of this method.
```suggestion
final Secret secret = findSecret(targetApiKey, targetCredentialsApiKey);
if (DatadogHttpClient.validateDefaultIntakeConnection(targetApiURL, secret)) {
      return FormValidation.ok("Great! Your API key is valid.");
  } else {
      return FormValidation.error("Hmmm, your API key seems to be invalid.");
  }

```

Reply:
Sure, I can look into that.

Reply:
Thanks for the suggestion! It helped clean up the code and allowed me to write more tests :)